extensions/smallrye-context-propagation/deployment/src/main/java/io/quarkus/smallrye/context/deployment/SmallRyeContextPropagationProcessor.java

```diff
@@ -86,4 +106,239 @@ void build(SmallRyeContextPropagationRecorder recorder,
                         .supplier(recorder.initializeManagedExecutor(executorBuildItem.getExecutorProxy()))
                         .setRuntimeInit().done());
     }
+
+    // transform IPs for ManagedExecutor/ThreadContext that use config annotation and don't yet have @NamedInstance
+    @BuildStep
+    InjectionPointTransformerBuildItem transformInjectionPoint() {
+        return new InjectionPointTransformerBuildItem(new InjectionPointsTransformer() {
+            @Override
+            public boolean appliesTo(Type requiredType) {
+                return true;
+            }
+
+            @Override
+            public void transform(TransformationContext transformationContext) {
+                // parse out the type being injected
+                DotName injectionPointType;
+                String mpConfigIpName;
+                AnnotationTarget target = transformationContext.getTarget();
+                final String nameDelimiter = "/";
+                switch (target.kind()) {
+                    case FIELD:
+                        injectionPointType = target.asField().type().name();
+                        mpConfigIpName = target.asField().declaringClass().name().toString()
+                                + nameDelimiter
+                                + target.asField().name();
+                        break;
+                    case METHOD_PARAMETER:
+                        injectionPointType = target.asMethodParameter().method().parameters()
+                                .get(target.asMethodParameter().position()).name();
+                        mpConfigIpName = target.asMethodParameter().method().declaringClass().name().toString()
+                                + nameDelimiter
+                                + target.asMethodParameter().method().name()
+                                + nameDelimiter
+                                + (target.asMethodParameter().position() + 1);
+                        break;
+                    // any other value is unexpected and we skip that
+                    default:
+                        return;
+                }
+                if (injectionPointType == null || !(injectionPointType.equals(DotNames.MANAGED_EXECUTOR)
+                        || injectionPointType.equals(DotNames.THREAD_CONTEXT))) {
+                    return;
+                }
+                // we don't care about any injection point that has custom qualifiers on it (inluding @NamedInstance)
+                if (transformationContext.getQualifiers().stream()
+                        .anyMatch(ann -> !ann.name().equals(DotNames.ANY)
+                                && !ann.name().equals(DotNames.DEFAULT))) {
```

Comment:
But this does not match the case when a `@NamedInstance` is declared, or?

Reply:
It doesn't and that's intended (and the comment above this code notes that).
Any IP that already has `@NamedInstance` doesn't need transformation - it has its own unique name preset by user.

Reply:
I see. Well, then it might make sense to move this check before the logic that derives `mpConfigIpName` above.

Reply:
Good point, now that I can check the required type earlier, I can move this one as well :)